### ISSUE TITLE
Add method for computing inverse

### ISSUE THREAD

**Issue by ndey96** [michaelsdr/momentumnet#19]
Hi thanks so much for your work!

Is it possible to add a method for computing the inverse of a layer's activations? Both the [iResNet](https://github.com/jhjacobsen/invertible-resnet) and [iRevNet](https://github.com/jhjacobsen/pytorch-i-revnet) repos have this method. Adding this would really help me experiment with your model more.

Thanks!

**Comment by isaacrob** [comment]
Hello, I am also interested in this!

**Comment by pierreablin** [comment]
Hi,

We will soon add a pr that implements this.

It is important to note that we cannot define a function `inverse(x)`, as we need the final value of the velocity `v` to recover the initial value of `x`. We can only define a function `inverse(x, v)`.

For instance, given an output `x`, we cannot really find y such that `net(y)= x`, because we would need to know in advance what the corresponding final velocity `v` would be.

In other words, the map x -> net(x) is not easily inverted while the map (x_0, v_0) -> (x_N, v_N) is easily inverted, but arguably much less interesting.
